### PR TITLE
Signatures: add resources with new URI scheme

### PR DESCRIPTION
**Pull request by fitzthum**
I think this is all we need, but I am not able to get the non-tee tests to pass locally even with upstream. The non-tee tests produce erratic results even when other tests run fine. We should test this with a Kata Agent built from https://github.com/kata-containers/kata-containers/pull/6404.

If you are in a timezone ahead of me feel free to do whatever you want with this PR.

@stevenhorsman @jepio

Fixes: #5576

### CODE REVIEW

**Comment by jepio** [comment]
@stevenhorsman i'm trying to run a test with the ttrpc PR here: https://github.com/kata-containers/kata-containers/pull/6569

**Comment by stevenhorsman** [comment]
> @stevenhorsman i'm trying to run a test with the ttrpc PR here: [kata-containers/kata-containers#6569](https://github.com/kata-containers/kata-containers/pull/6569)

Nice - it looks like you've got the depends-on correct :)

**Comment by jepio** [comment]
cosign tests were still failing, so i pushed an extra commit to #5579

File: integration/confidential/fixtures/offline-fs-kbc/s390x/aa-offline_fs_kbc-resources.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
...
+  "default/security-policy/test": "ewogICAgImRlZmF1bHQiOiBbCiAgICAgICAgewogICAgICAgICAgICAidHlwZSI6ICJpbnNlY3VyZUFjY2VwdEFueXRoaW5nIgogICAgICAgIH0KICAgIF0sCiAgICAidHJhbnNwb3J0cyI6IHsKICAgICAgICAiZG9ja2VyIjogewogICAgICAgICAgICAicXVheS5pby9rYXRhLWNvbnRhaW5lcnMvY29uZmlkZW50aWFsLWNvbnRhaW5lcnMiOiBbCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgInR5cGUiOiAic2lnbmVkQnkiLAogICAgICAgICAgICAgICAgICAgICJrZXlUeXBlIjogIkdQR0tleXMiLAogICAgICAgICAgICAgICAgICAgICJrZXlQYXRoIjogIi9ldGMvY29udGFpbmVycy9xdWF5X3ZlcmlmaWNhdGlvbi9wdWJsaWMuZ3BnIgogICAgICAgICAgICAgICAgfQogICAgICAgICAgICBdCiAgICAgICAgfQogICAgfQp9Cg==",
```
Comment (Xynnn007) [inline integration/confidential/fixtures/offline-fs-kbc/s390x/aa-offline_fs_kbc-resources.json:5]: This policy is
```json
{
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports": {
        "docker": {
            "quay.io/kata-containers/confidential-containers": [
                {
                    "type": "signedBy",
                    "keyType": "GPGKeys",
                    "keyPath": "/etc/containers/quay_verification/public.gpg"
                }
            ]
        }
    }
}
```

That means the GPG keyring  is located inside the guest image. If the test aims to fetch the GPG keyring  from KBS, please use the following
```
ewogICAgImRlZmF1bHQiOiBbCiAgICAgICAgewogICAgICAgICAgICAidHlwZSI6ICJpbnNlY3VyZUFjY2VwdEFueXRoaW5nIgogICAgICAgIH0KICAgIF0sCiAgICAidHJhbnNwb3J0cyI6IHsKICAgICAgICAiZG9ja2VyIjogewogICAgICAgICAgICAicXVheS5pby9rYXRhLWNvbnRhaW5lcnMvY29uZmlkZW50aWFsLWNvbnRhaW5lcnMiOiBbCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgInR5cGUiOiAic2lnbmVkQnkiLAogICAgICAgICAgICAgICAgICAgICJrZXlUeXBlIjogIkdQR0tleXMiLAogICAgICAgICAgICAgICAgICAgICJrZXlQYXRoIjogImticzovLy9kZWZhdWx0L2dwZy1wdWJsaWMtY29uZmlnL3Rlc3QiCiAgICAgICAgICAgICAgICB9CiAgICAgICAgICAgIF0KICAgICAgICB9CiAgICB9Cn0=
```

which means
```json
{
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports": {
        "docker": {
            "quay.io/kata-containers/confidential-containers": [
                {
                    "type": "signedBy",
                    "keyType": "GPGKeys",
                    "keyPath": "kbs:///default/gpg-public-config/test"
                }
            ]
        }
    }
}
```

File: integration/confidential/fixtures/offline-fs-kbc/s390x/aa-offline_fs_kbc-resources.json [hunk 2]
```diff
@@ -1,5 +1,8 @@
...
...0k1YlJvcStKeUkrWUh4dm9iaTlxZkVRT3lKVk1JNgpMb294ZlVaWkFEN0UzWkkwdzhld296UmtWV2tnRktzSmVVTko5RzVZK1RFSFdSUG45TlVFNzEyVmxvNWhVT04zCkNUQ0N2ek5CQmcycTRHQ1VFMno4WG1QRkRwMzJMZ1d1em9HWVJVdEh3SmVKTmhZQWtGMVpIYmJDczF0UExkTWIKTzZJaW8zVVVnRHFmRy9TOVJpOGhJTHcyVmx3eVBFQSs1YitXd2JUbTBiL3ZPMUUvcDVvRlFLaDNvd1MrcVNMbgo4TnQ5Y2pQSGJDNUIwL0VBRVFFQUFZa0J2QVFZQVFvQUpoWWhCS1RqSGkvOTdzYWlEaWZCeDZ1T1ZxczU2eFZqCkJRSmpkbDRCQWhzTUJRa0R3bWNBQUFvSkVLdU9WcXM1NnhWajNqb0wrd1lUVy9BU3hGRXVGUWJvZm1PbTh5OGEKUUl2TWpiRDBnQUlTM0Y5QytxUnlSUjYyN1dPak5jSGl3RVA5YUh3SVZFRXZiaVJkZk9lNVV2Z2M4NlpjaEFhMwo1Yk9XR3VKQ0JIWU5GNU9RcUpxTXlvWmpmRWRFSEpWVW9wbGhOd3NUa0V3MGc3ZXU0TEpkMWxsN0UvVVpDWmVvClhudlJxTVFEUjRUV2ZYeks5dzZONXNrZHJXcHZWeGNCNEUwVW9BVmh2alVNalo2S2ZmUGYrOS9SWTRLdG83S0EKbXVhZUkxV0M4RGRjZUJnZDg5QXNhMVVOWCtpVC9DVGw0endQWXQzMjVjcmhIVG5YZ05BRFo4QXZhUUFyc24wWQpGT0svcGJ0aUczOEtkalJBQjNlN3dDTFdWaVV4QkhBQ1lyUzkvUjBxVzRCa1Y5OHNwZXRSVWxhYlVycDVFbHNaClNXTVFpNk95NzlYRkZMam55eS9JbzdGSDREdXlKQTlLZ3lyaDFJUXU0c3ZWeWhaTWVGbW5GUkdkSUVqM2JBcW0Kbm9ZSlA2WElTeUFGNjB1UEJteEdidXpvTm9CbUVKZDFHeTVBZ0ZyQ1p3K1gxc1ZtTVZUMit3YVpvK0lINjlGVwpoNW1hdERZdDh2VE1zRmxuaEN5NllmUDdQcEhaQmNBa1JXYU9SNkVrL1E9PQo9aWpVOQotLS0tLUVORCBQR1AgUFVCTElDIEtFWSBCTE9DSy0tLS0tCg=="
```
Comment (Xynnn007) [inline integration/confidential/fixtures/offline-fs-kbc/s390x/aa-offline_fs_kbc-resources.json:7]: the key name should be `"default/gpg-public-config/test"`

File: integration/confidential/fixtures/offline-fs-kbc/x86_64/aa-offline_fs_kbc-resources.json [hunk 3]
```diff
@@ -1,5 +1,8 @@
 {
   "Policy": "ewogICAgImRlZmF1bHQiOiBbCiAgICAgICAgewogICAgICAgICAgICAidHlwZSI6ICJpbnNlY3VyZUFjY2VwdEFueXRoaW5nIgogICAgICAgIH0KICAgIF0sCiAgICAidHJhbnNwb3J0cyI6IHsKICAgICAgICAiZG9ja2VyIjogewogICAgICAgICAgICAicXVheS5pby9rYXRhLWNvbnRhaW5lcnMiOiBbCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgInR5cGUiOiAic2lnbmVkQnkiLAogICAgICAgICAgICAgICAgICAgICJrZXlUeXBlIjogIkdQR0tleXMiLAogICAgICAgICAgICAgICAgICAgICJrZXlQYXRoIjogIi9ldGMvY29udGFpbmVycy9xdWF5X3ZlcmlmaWNhdGlvbi9wdWJsaWMuZ3BnIgogICAgICAgICAgICAgICAgfQogICAgICAgICAgICBdCiAgICAgICAgfQogICAgfQp9Cg==",
```
Comment (Xynnn007) [inline integration/confidential/fixtures/offline-fs-kbc/x86_64/aa-offline_fs_kbc-resources.json:2]: Also, we do not need `"Policy"`, `"Sigstore Config"` and `"GPG Keyring"` anymore

File: integration/confidential/fixtures/offline-fs-kbc/x86_64/aa-offline_fs_kbc-resources.json [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+  "default/security-policy/test": "ewogICAgImRlZmF1bHQiOiBbCiAgICAgICAgewogICAgICAgICAgICAidHlwZSI6ICJpbnNlY3VyZUFjY2VwdEFueXRoaW5nIgogICAgICAgIH0KICAgIF0sCiAgICAidHJhbnNwb3J0cyI6IHsKICAgICAgICAiZG9ja2VyIjogewogICAgICAgICAgICAicXVheS5pby9rYXRhLWNvbnRhaW5lcnMiOiBbCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgInR5cGUiOiAic2lnbmVkQnkiLAogICAgICAgICAgICAgICAgICAgICJrZXlUeXBlIjogIkdQR0tleXMiLAogICAgICAgICAgICAgICAgICAgICJrZXlQYXRoIjogIi9ldGMvY29udGFpbmVycy9xdWF5X3ZlcmlmaWNhdGlvbi9wdWJsaWMuZ3BnIgogICAgICAgICAgICAgICAgfQogICAgICAgICAgICBdCiAgICAgICAgfQogICAgfQp9Cg==",
```
Comment (Xynnn007) [inline integration/confidential/fixtures/offline-fs-kbc/x86_64/aa-offline_fs_kbc-resources.json:5]: Also, here the policy is
```json
{
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports": {
        "docker": {
            "quay.io/kata-containers": [
                {
                    "type": "signedBy",
                    "keyType": "GPGKeys",
                    "keyPath": "/etc/containers/quay_verification/public.gpg"
                }
            ]
        }
    }
}
```

That means the GPG keyring  is located inside the guest image. If the test aims to fetch the GPG keyring  from KBS, please use the following
```
ewogICAgImRlZmF1bHQiOiBbCiAgICAgICAgewogICAgICAgICAgICAidHlwZSI6ICJpbnNlY3VyZUFjY2VwdEFueXRoaW5nIgogICAgICAgIH0KICAgIF0sCiAgICAidHJhbnNwb3J0cyI6IHsKICAgICAgICAiZG9ja2VyIjogewogICAgICAgICAgICAicXVheS5pby9rYXRhLWNvbnRhaW5lcnMiOiBbCiAgICAgICAgICAgICAgICB7CiAgICAgICAgICAgICAgICAgICAgInR5cGUiOiAic2lnbmVkQnkiLAogICAgICAgICAgICAgICAgICAgICJrZXlUeXBlIjogIkdQR0tleXMiLAogICAgICAgICAgICAgICAgICAgICJrZXlQYXRoIjogImticzovLy9kZWZhdWx0L2dwZy1wdWJsaWMtY29uZmlnL3Rlc3QiCiAgICAgICAgICAgICAgICB9CiAgICAgICAgICAgIF0KICAgICAgICB9CiAgICB9Cn0=
```

which means
```json
{
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports": {
        "docker": {
            "quay.io/kata-containers": [
                {
                    "type": "signedBy",
                    "keyType": "GPGKeys",
                    "keyPath": "kbs:///default/gpg-public-config/test"
                }
            ]
        }
    }
}
```

File: integration/confidential/fixtures/offline-fs-kbc/aa-offline_fs_kbc-resources.json.in [hunk 5]
```diff
@@ -3,5 +3,10 @@
   "Sigstore Config": "${SIGSTORE_CONFIG}",
   "GPG Keyring": "${GPG_KEYRING}",
   "Cosign Key": "${COSIGN_KEY}",
-  "Credential": "${CREDENTIAL}"
-}
\ No newline at end of file
+  "Credential": "${CREDENTIAL}",
```
[the remark this answers is not in the document]
Comment (jepio) [reply integration/confidential/fixtures/offline-fs-kbc/aa-offline_fs_kbc-resources.json.in:6]: these are for compatibility with AA before KBS URI support. The goal is to be able to merge this PR before the ttrpc switch.

**Pull request merged**